### DISCUSSION TITLE
v2.0 beta - phinger licking good

### DISCUSSION THREAD

**Discussion by phisch** [phisch/phinger-cursors#15] [Beta Testing]
This is the first release, that uses the new [Cursor Theme Builder](https://github.com/phisch/cursor-theme-builder). It brings plenty advantages over the last one, like an animation system and support to build left-handed cursors. Check out the cursor theme builder for more details.

## Sexy as always
It was a long journey to get this release out, mainly due to writing a new build system. But hopefully, you can all agree, that the new phingers are sexier than ever:

![preview](https://github.com/phisch/phinger-cursors/assets/1282767/2144e929-837d-43e0-92d2-4b0c7a6e61a8)

## New Borders
Unlike traditional cursors, phingers now come with a translucent border, which provides enough contrast, but isn't too much.

![borders](https://github.com/phisch/phinger-cursors/assets/1282767/69615c55-6f09-4b78-9b72-cbb3fb4cec14)

## Space efficiency and new badges
Old phingers had to fight with very limited space, since they used a shadow and borders everywhere. The shadow did not provide enough benefits to justify the lost space, so they were now traded in for that lost space! Additionally, elements that provide enough contrast themselves, like badges, don't use a border anymore, which made it possible to increase the badge size significantly. All badge icons were updated to make better use of their new space.

![badges](https://github.com/phisch/phinger-cursors/assets/1282767/8c26b970-0e95-4845-a4f1-ef0a436f8b6e)

## Love For Lefties
The new Cursor Theme Builder supports transforming right handed cursors into left handed cursors. This means, this release officially comes with left handed variants. Instead of using `phinger-cursors-light` or `phinger-cursors-dark`, use `phinger-cursors-light-left` or `phinger-cursors-dark-left`.

![left handed](https://github.com/phisch/phinger-cursors/assets/1282767/1d30fc46-1955-4218-9493-c3f71db25fe0)

## 100% Wayland (and CSS3)
A lot has happened since the last release over 2 years ago. Wayland, which is almost everywhere the default now, has introduced the [cursor-shape-v1 protocol](https://wayland.app/protocols/cursor-shape-v1) which defines an [enum of cursors](https://wayland.app/protocols/cursor-shape-v1#wp_cursor_shape_device_v1:enum:shape) which was taken from the [CSS spec](https://w3c.github.io/csswg-drafts/css-ui/#cursor).

Phingers now comes with 100% compatibility to this standard, which makes it future proof.
![wayland set](https://github.com/phisch/phinger-cursors/assets/1282767/002e55c5-47fb-481b-978c-bf2908f72238)

## Taking Out The Trash
Some of the cursors in older releases were simply not used by any application. Most of those cursors had to say goodbye, and are now happily living in the trash, where they belong:

 - all border cursors
 - single axis scroll cursors
 - x11 cursor
 - wayland cursor

Nobody will miss those!

## Aliases
All of the currently open issues are due to missing aliases. Since core cursors follow the cursor-shape-v1 protocol, aliases have gotten a fresh start as well. Most commonly used aliases should now be set properly. But I'm happy to go through open issues and get some feedback from beta testers to figure out if there are still inconsistencies.

## New Build System
The main reason why this release took so long, was because I was working on an entirely [new build system](https://github.com/phisch/cursor-theme-builder). It has now gotten its first release, and works well enough for phinger-cursors to switch to it. It has a pretty okay [animation system](https://phisch.github.io/cursor-theme-builder/), which makes key-framing animations a thing of the past.

<hr /><em>This discussion was created from the release <a href='https://github.com/phisch/phinger-cursors/releases/tag/'>v2.0 beta - phinger licking good</a>.</em>

**Comment by haedrichowen** [comment]
Very exciting!!

**Comment by Lucki** [comment]
I have version 2 now a few weeks in use. I'm using dark mode everywhere and since the update to version 2 I'm often searching my cursor on the screen - the outline contrast is too low to see the cursor reliably. I didn't experience this with the version 1 cursors.

When I move my cursor to the picture above in the post above the outline is significantly darker than the picture is showing. I tried to make a screenshot, but on the screenshot they're identical.

Screenshot v2 vs v1:
![image](https://github.com/phisch/phinger-cursors/assets/1408843/4d135d5f-6f0c-4edf-834e-d1662cd69c59)![image](https://github.com/phisch/phinger-cursors/assets/1408843/05bf5ac8-847e-4a2f-bf02-0f1bd1ed1c79)

"Screenshot" v2 vs v1:
<img src="https://github.com/phisch/phinger-cursors/assets/1408843/74181414-3a58-47a1-9da2-a7ccd815abcb" width="158" /><img src="https://github.com/phisch/phinger-cursors/assets/1408843/a23cb8b2-0260-458e-8c86-e8624ef41837" width=158 />

**Reply by phisch** [reply]
The screenshot shows what it actually is. It seems like there is maybe some scaling happening on your system. What's your desktop environment, display settings, and how big is your monitor?

I've been very happy with the new, slightly translucent outlines. But I do recommend using the light cursor when using a dark theme. That's the intended use, though not necessarily what you have to do!

**Reply by Lucki** [reply]
> What's your desktop environment, display settings, and how big is your monitor?

Gnome 46.1, one 1440p (27") and one 1080p (24") monitor - same low contrast cursor on both monitors, no scaling on both. Not sure what's meant with the requested display settings, sry.

> But I do recommend using the light cursor when using a dark theme.

Well, it was fine with version 1.
I looked in the issue section but saw no one complaining, so I'm not sure if it's a "me" problem and posted in this discussion.

---

Here's version 1 for comparison (without a restart of the gnome shell):
![image](https://github.com/phisch/phinger-cursors/assets/1408843/05bf5ac8-847e-4a2f-bf02-0f1bd1ed1c79)
<img src="https://github.com/phisch/phinger-cursors/assets/1408843/a23cb8b2-0260-458e-8c86-e8624ef41837" width=158 />

**Reply by phisch** [reply]
So it doesn't look like it's a scaling issue then. :thinking: 

The border contrast was lowered on purpose, and I don't intend to raise that contrast again. The cursor should contrast enough on its own, and the border just plays a supporting role.

I did encounter some very weird behavior with some apps and toolkits that, for some reason, request the wrong cursor size and scale it themselves. That could possibly explain the slightly too low contrast in your photographed screenshot. But I was a bit burned out on this project after the last release, so I didn't really want to dive into how GTK, chromium, some Wayland compositors and others actually load, scale, and render the cursor sprites.

From your screenshot, it looks like you used the 24px base sprite. You could try to see if the same issue happens when using the 32px base sprite by setting your cursor size to 32 instead of 24.

But unfortunately, I can't provide a solution to your problem at the moment, I'm sorry! :frowning_face:

**Reply by Lucki** [reply]
> The border contrast was lowered on purpose, and I don't intend to raise that contrast again. The cursor should contrast enough on its own, and the border just plays a supporting role.

Looking at the intended pictures I think it would be fine but it's way darker, that's the strange part. This isn't a request to change anything!

> From your screenshot, it looks like you used the 24px base sprite. You could try to see if the same issue happens when using the 32px base sprite by setting your cursor size to 32 instead of 24.

I never did such thing, I only selected the cursor in the gnome tweak tool. Don't know how I could choose the size directly there.

> But unfortunately, I can't provide a solution to your problem at the moment, I'm sorry! ☹️

Don't worry. I currently switched to the light variant and see how I like that one. If I don't, then I'll simply switch to back the version 1. I'm sorry!

**Reply by phisch** [reply]
I believe you can run `gsettings set org.gnome.desktop.interface cursor-size 64` to set the cursor size in GNOME.

**Reply by Lucki** [reply]
Yes, that changes the size, the outline still appears darker than in the example image, although less noticeable - strange. Thanks anyway!

<img src="https://github.com/phisch/phinger-cursors/assets/1408843/533d886b-6a8d-4693-9d91-5f05d82fdd4e" width=158 />

**Reply by Lucki** [reply]
I wanted to report back to this.
With the slight outline boost in the 2.1 version the perceptive outline brightness now matches the given preview examples.
And as I expected previously ("Looking at the intended pictures I think it would be fine") it's actually fine for me now - I'm not loosing track of the cursor in v2.1.

Screenshot v2.1 vs "Screenshot" v2.1
![grafik](https://github.com/user-attachments/assets/9514087d-a884-4378-8413-9e4803ca7315)!<img src="https://github.com/user-attachments/assets/6b2eab3d-e67c-4df4-87a3-6a671f4b9d01" width="158" />
Screenshot v2.0 vs "Screenshot" v2.0
![image](https://github.com/phisch/phinger-cursors/assets/1408843/4d135d5f-6f0c-4edf-834e-d1662cd69c59)<img src="https://github.com/phisch/phinger-cursors/assets/1408843/74181414-3a58-47a1-9da2-a7ccd815abcb" width="158" />
Screenshot v1 vs "Screenshot" v1
![image](https://github.com/phisch/phinger-cursors/assets/1408843/05bf5ac8-847e-4a2f-bf02-0f1bd1ed1c79)<img src="https://github.com/phisch/phinger-cursors/assets/1408843/a23cb8b2-0260-458e-8c86-e8624ef41837" width=158 />

**Reply by phisch** [reply]
I'm glad to hear that!
